classes/mail/mailables/Recipient.inc.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * @file mail/mailables/Recipient.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Recipient
+ * @ingroup mail_mailables
+ *
+ * @brief mailable's trait to support User recipients
+ */
+
+namespace PKP\mail\mailables;
+
+use BadMethodCallException;
+use InvalidArgumentException;
+use PKP\mail\Mailable;
+use PKP\user\User;
+
+trait Recipient {
+
+    /**
+     * @copydoc Illuminate\Mail\Mailable::setAddress()
+     */
+    abstract protected function setAddress($address, $name = null, $property = 'to');
+
+    /**
+     * @copydoc PKP\mail\Mailable::addVariables()
+     */
+    abstract public function addVariables(array $variables) : Mailable;
+
+    /**
+     * @copydoc PKP\mail\Mailable::templateVariablesMap()
+     */
+    abstract static protected function templateVariablesMap() : array;
```

Comment:
Basically, an abstract method in the trait may also mean that it may be implemented in one of the parents. I haven't dig deeper though. But with recent changes in the variables map, it's not needed and should be removed.